src/ansys/fluent/core/session_pure_meshing.py

```diff
@@ -117,17 +117,17 @@ def workflow(self):
 
     def watertight(self):
         """Get a new watertight workflow."""
-        self._base_meshing.watertight_workflow.reinitialize()
+        self._base_meshing.watertight_workflow.initialize()
```

Comment:
Just curious, is there any benefit in returning the previous instance of the `WatertightMeshingWorkflow` class? If `watertight()` always creates a new instance of `WatertightMeshingWorkflow`, we could clear the root-affected callback at the `WatertightMeshingWorkflow.__init__` method.

Reply:
Yes, the current code seems unnecessarily complicated. Why is it telling objects to initialize and reinitialize etc?

Reply:
We can discuss this once in today's meeting.

Reply:
@mkundu1, Please put in a github issue for refactoring this. Thank you